### PR TITLE
Check in Cargo.lock

### PR DESCRIPTION
**Pull request by mzabaluev**
Almost all the build targets in the Cargo workspace are end artefacts
and therefore need to reference state of the build in version control.

Also add cargo dependency caching to the CI workflow now that we have `Cargo.lock` content to easily use as the cache key.

### CODE REVIEW

File: .github/workflows/ci_checks.yaml [hunk 1]
```diff
@@ -99,21 +175,6 @@ jobs:
       - name: Test in firefox
         run: wasm-pack test --headless --firefox ./bindings/wallet-js
 
-  test-key-gen-lib:
```
Comment (Zeegomo) [inline .github/workflows/ci_checks.yaml:102]: Why was this removed?

Comment (mzabaluev) [reply .github/workflows/ci_checks.yaml:102]: This js library is no longer needed by Daedalus, if it ever used at all; I think the developers figured out how to generate ed25519 keys without a weird Rust library to help.

Comment (enzoc4) [reply .github/workflows/ci_checks.yaml:102]: should we also remove the actual code?

**Comment by enzoc4** [comment]
I wonder how will this affect the testing crates that are using this library. Are those able to ignore the `Cargo.lock` here? Or they would run into problems with multiple versions of `chain-libs`?

File: .github/workflows/ci_checks.yaml (hunk 1 above)
Comment (mzabaluev) [reply .github/workflows/ci_checks.yaml:102]: Nobody's asking for a new release of the keygen package, so I think it's time.

**Comment by mzabaluev** [comment]
As far as I know, the `Cargo.lock` from dependency library's workspace has no bearing on projects that use the library.

**Review by enzoc4 — APPROVED** [review]
